pkg/admin/handler/traffic_condition_rule.go

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package handler
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+import (
+	mesh_proto "github.com/apache/dubbo-kubernetes/api/mesh/v1alpha1"
+	"github.com/apache/dubbo-kubernetes/pkg/admin/model"
+	"github.com/apache/dubbo-kubernetes/pkg/core/consts"
+	"github.com/apache/dubbo-kubernetes/pkg/core/resources/apis/mesh"
+	res_model "github.com/apache/dubbo-kubernetes/pkg/core/resources/model"
+	"github.com/apache/dubbo-kubernetes/pkg/core/resources/store"
+	core_runtime "github.com/apache/dubbo-kubernetes/pkg/core/runtime"
+)
+
+func ConditionRuleSearch(rt core_runtime.Runtime) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		resList := &mesh.ConditionRouteResourceList{
+			Items: make([]*mesh.ConditionRouteResource, 0),
+		}
+		if err := rt.ResourceManager().List(rt.AppContext(), resList); err != nil {
+			c.JSON(http.StatusBadRequest, model.NewErrorResp(err.Error()))
+			return
+		}
+		resp := model.ConditionRuleSearchResp{
+			Code:    http.StatusOK,
+			Message: "success",
+			Data:    make([]model.ConditionRuleSearchResp_Data, 0, len(resList.Items)),
+		}
+		for _, item := range resList.Items {
+			resp.Data = append(resp.Data, model.ConditionRuleSearchResp_Data{
+				RuleName:   item.Meta.GetName(),
+				Scope:      item.Spec.GetScope(),
+				CreateTime: item.Meta.GetCreationTime().String(),
+				Enabled:    item.Spec.GetEnabled(),
+			})
+		}
+		c.JSON(http.StatusOK, resp)
+	}
+}
+
+func GetConditionRuleWithRuleName(rt core_runtime.Runtime) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var appName string
+		ruleName := c.Param("ruleName")
+		res := &mesh.ConditionRouteResource{Spec: &mesh_proto.ConditionRoute{}}
+		if strings.HasSuffix(ruleName, consts.ConditionRuleSuffix) {
+			appName = ruleName[:len(ruleName)-len(consts.ConditionRuleSuffix)]
```

Comment:
`ruleName` does not always has `appName.suffix` pattern, it could be `interfaceName.suffix`.

Reply:
Both the following formats are legal, `org.apache.dubbo.samples.UserService::.condition-router` and `shop-user.conditionrule`